api/db/populate-notifications.js

```diff
@@ -0,0 +1,222 @@
+  const notification = new Notification({
+    user: notificationParams.user,
+    type: notificationParams.type,
+    read: notificationParams.read,
+  });
+  if (notification.read) notification.readAt = new Date();
+  if (notificationParams.referencedUser) {
+    notification.referencedUser = notificationParams.referencedUser;
+  }
+  if (notificationParams.referencedEvent) {
+    notification.referencedEvent = notificationParams.referencedEvent;
+  }
+  if (notificationParams.referencedFriendship) {
+    notification.referencedFriendship = notificationParams.referencedFriendship;
+  }
+
+  await notification.save();
+
+  if (displayLogs) {
+    console.log('\n', '\x1b[0m', `New notification created: ${notification}`);
+  }
+  return notification;
+};
+
+const createNotifications = async (verbose) => {
+  console.log('\n', '\x1b[0m', 'Populating notifications collection...');
+  displayLogs = verbose;
+
+  if ((await Notification.countDocuments()) !== 0) {
+    const proceed = await createPrompt(
+      'The notifications collection already exists. If you continue, the existing collection will be dropped to avoid duplication. Do you want to continue?'
+    );
+
+    if (proceed) {
+      await Notification.collection.drop();
+      console.log('\x1b[31m', 'Dropped old notifications collection');
+    } else {
+      console.log('\x1b[33m', 'Aborted notifications collection population');
+      return;
+    }
+  }
+
+  const notificationsArray = [];
+  const usersList = await User.find();
+  const eventsList = await Event.find();
+  const attendantsList = await Attendant.find();
+  const commentsList = await Comment.find();
+  const friendshipsList = await Friendship.find();
+  const messagesList = await Message.find();
+
+  // For each user
+  for (const user of usersList) {
+    let read = user.emailConfirmed;
+    // Email confirmation notification
+    notificationsArray.push(
+      await createNotification({
+        user,
+        type: types[0],
+        read,
+      })
+    );
+    // Create event notification
+    read = eventsList.some(
+      (event) => event.createdBy.toString === user._id.toString()
+    );
+    notificationsArray.push(
+      await createNotification({
+        user,
+        type: types[1],
+        read,
+      })
+    );
+    // Create upload avatar notification
+    read = user.avatar ? true : false;
+    notificationsArray.push(
+      await createNotification({
+        user,
+        type: types[2],
+        read,
+      })
+    );
+    // Selected interests notification
+    const interests = await Interest.find({ user });
+    read = interests.length > 0;
+    notificationsArray.push(
+      await createNotification({
+        user,
+        type: types[3],
+        read,
+      })
+    );
+
+    // Create attendee and comment notifications
+    const usersEvents = eventsList.filter(
+      (event) => event.createdBy.toString() === user._id.toString()
+    );
```

Comment:
As above we are using an `eventsList.some(...)` with the same condition, maybe it would make sense to move this `eventsList.filter(...)` above and, to get the needed boolean value, just check if the `usersEvents.length > 0` instead of using the `some`.

Also: should be `userEvents`?